xcdat/dataset.py

```diff
@@ -308,16 +313,24 @@ def decode_non_cf_time(dataset: xr.Dataset) -> xr.Dataset:
     time_bounds = ds.get(time.attrs.get("bounds"), None)
     units_attr = time.attrs.get("units")
 
-    # If the time units cannot be split into a unit and reference date, it
-    # cannot be decoded so the original dateset is returned.
     try:
         units, ref_date = _split_time_units_attr(units_attr)
     except ValueError:
+        logger.warning(
+            "Attempted to decode non-CF compliant time coordinates, but the units "
+            f"({units}) is not in a supported format ('months since...' or "
+            "'months since...'"
+        )
         return ds
 
-    ref_date = pd.to_datetime(ref_date)
+    # Create a reference date object and use the relative delta offsets
+    # to create `cftime.datetime` objects. Note, the reference date object type
+    # must start as `datetime` because `cftime` does not support arithmetic with
+    # `rd.relativedelta`.
+    ref_dt_obj = datetime.strptime(ref_date, "%Y-%m-%d")
```

Comment:
I'm not sure what to do here...

Older LLNL datasets/code used time units like "months since 1800". `pd.to_datetime(ref_date).to_pydatetime()` will handle this, but `datetime.strptime(ref_date, "%Y-%m-%d")` does not.

`pd.to_datetime` also seems to be able to figure out how to correctly interpret other `ref_date` values, e.g., '1/1/1800`. But it does not work for very large years (e.g., "1/1/3000").

Reply:
Though this seems to help: `pd.to_datetime(ref_date, errors='coerce').to_pydatetime()`

Reply:
What about the following:

```
from dateutil import parser

ref_dt_obj = parser.parse(ref_date, default=datetime.datetime(2000, 1, 1))
```

Reply:
I just tried `parser.parse` on different reference date formats.

It doesn't seem to be defaulting to 1 as the placeholder value for month or day.

```python
$ parser.parse("1800")
datetime.datetime(1800, 7, 28, 0, 0)

$ parser.parse("1800-01")
datetime.datetime(1800, 1, 28, 0, 0)

$ parser.parse("1800-01-01")
datetime.datetime(1800, 1, 1, 0, 0)
```

For the units, does it sometimes extend beyond day (e.g., "months since 2000-01-01T00:00:00.000000000")?

If `parser.parse` doesn't work and there's no other sufficient datetime string parser, we can write our own function.

Reply:
Actually, specifying "default=" works for the placeholders.

```python
$ parser.parse("1800", default=datetime.datetime(2000, 1, 1))
datetime.datetime(1800, 1, 1, 0, 0)

$ parser.parse("1800-01", default=datetime.datetime(2000, 1, 1))
datetime.datetime(1800, 1, 1, 0, 0)

$ parser.parse("1800-01-01", default=datetime.datetime(2000, 1, 1))
datetime.datetime(1800, 1, 1, 0, 0)
```

**RE: For the units, does it sometimes extend beyond day (e.g., "months since 2000-01-01T00:00:00.000000000")?**
It also doesn't matter if the units extend beyond day, since all we care about is having the day at minimum in the datetime object.